File: sprint3/By languages/C++/lesson2code2.cpp
#include <iostream>

void stairs_builder(int n) {
    if (n == 0) {
        return;
    }
    // build 1 step
    std::cout << "Осталось построить " << n << " ступеней." << std::endl;
    stairs_builder(n - 1);
}

int main() {
    stairs_builder(5);
    return 0;
}

File: sprint3/By languages/go/lesson2code2.go
package main

import "fmt"

func stairsBuilder(n int) {
    if n == 0 {
        return
    }
    // build 1 step
    fmt.Printf("Осталось построить %d ступеней.\n", n)
    stairsBuilder(n - 1)
}

func main() {
    stairsBuilder(5)
}
